agent/api/task/task.go

```diff
@@ -3502,3 +3504,138 @@ func (task *Task) IsServiceConnectConnectionDraining() bool {
 	defer task.lock.RUnlock()
 	return task.ServiceConnectConnectionDrainingUnsafe
 }
+
+// ToHostResources will convert a task to a map of resources which ECS takes into account when scheduling tasks on instances
+// * CPU
+//   - If task level CPU is set, use that
+//   - Else add up container CPUs
+//
+// * Memory
+//   - If task level memory is set, use that
+//   - Else add up container level
+//   - If memoryReservation field is set, use that
+//   - Else use memory field
+//
+// * Ports (TCP/UDP)
+//   - Only account for hostPort
+//   - Don't need to account for awsvpc mode, each task gets its own namespace
+//
+// * GPU
+//   - Return num of gpus requested (len of GPUIDs field)
+//
+// TODO remove ToHostResources is used
```

Comment:
nit: not sure what this TODO tells us.

Reply:
remove `when` ToHostResources is used, `when` got lost somewhere, will update it